src/collections/componentSet.ts

```diff
@@ -369,7 +369,17 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
    * @returns Object representation of a package manifest
    */
   public async getObject(destructiveType?: DestructiveChangesType): Promise<PackageManifestObject> {
-    const version = this.sourceApiVersion ?? this.apiVersion ?? `${await getCurrentApiVersion()}.0`;
+    let version: string;
+    try {
+      version =
+        this.sourceApiVersion ??
+        this.apiVersion ??
+        process.env.SF_ORG_API_VERSION ??
```

Comment:
would you also consider using ConfigAggregator here?  (it would check local/global config AND the env)?  Like that would be to check local FS before making a callout?